src/Link/LinkProcessor.php

```diff
@@ -35,7 +34,30 @@ public static function preRenderLink($element) : array {
           $element['#attributes']['data-protocol'] = $scheme;
         }
       }
-      $element['#title']['#attributes'] = $element['#attributes'] ?? [];
+
+      if (!empty($element['#attributes']['data-is-external'])) {
```

Comment:
I think this can be simplified by moving `$use_helfi_link = TRUE` inside previous if condition:

```php
      if ($resolver->isExternal($element['#url'])) {
        $element['#attributes']['data-is-external'] = 'true';

        if ($scheme = $resolver->getProtocol($element['#url'])) {
          $element['#attributes']['data-protocol'] = $scheme;
        }
        $use_helfi_link = TRUE;
      }
```

Reply:
@tuutti yes, but I wasn't sure if there's any case where the  $element['#attributes'] 'data-protocol' or 'data-is-external' could come already populated from the render array for an internal link !? cause basically if that's the case, then twig conditions would already cover that case...
I can do it like that if you think it should not be the case that an internal link could have these data- attributes